pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -84,6 +86,32 @@ func (c *CSIControllerSet) WithManagementStateController(operandName string, sup
 	return c
 }
 
+// WithConditionalStaticResourcesController returns a *ControllerSet with a conditional static resources controller initialized.
+func (c *CSIControllerSet) WithConditionalStaticResourcesController(
+	name string,
+	kubeClient kubernetes.Interface,
+	dynamicClient dynamic.Interface,
+	kubeInformersForNamespace v1helpers.KubeInformersForNamespaces,
+	manifests resourceapply.AssetFunc,
+	files []string,
+	shouldCreateFnArg, shouldDeleteFnArg resourceapply.ConditionalFunction,
+) *CSIControllerSet {
+	c.conditionalStaticResourcesController = staticresourcecontroller.NewStaticResourceController(
```

Comment:
you either want this to be a slice or you want a way to stop people from calling it multple times.

Reply:
Yeah, I'd appreciate if  CSIControllerSet had `[]Controller` instead of fields. Worth some refactoring in the next release.

Reply:
IIRC the original idea was to prevent users from creating duplicate controllers